source/slang/slang-ast-expr.h

```diff
@@ -541,6 +541,14 @@ class TreatAsDifferentiableExpr : public Expr
 
     Expr* innerExpr;
     Scope* scope;
+    
+    enum Flavor 
+    {
```

Comment:
May need some comment here explaining the meaning of each value.

Reply:
Added some comments 👍